cql/src/main/java/org/apache/cassandra/stargate/transport/internal/messages/ResultMessage.java

```diff
@@ -375,7 +376,9 @@ public Result decode(ByteBuf body, ProtocolVersion version) {
       if (version.isGreaterThan(ProtocolVersion.V1))
         resultMetadata = RowsSubCodec.METADATA_CODEC.decode(body, version);
 
-      return new Result.Prepared(id, resultMetadataId, resultMetadata, metadata);
+      boolean isIdempotent = byteToBoolean(CBUtil.readBytes(body));
```

Comment:
Why does this add the idempotent flag to the native protocol?

Reply:
To propagate info about idempotency, we need to have it in the `Result.Prepared`.
Unfortunately, it means that a new instance of `Result.Prepared` needs to be propagated here: 
https://github.com/stargate/stargate/pull/1194/files#diff-85e0db2983367de622d1f158ed4ee09146dc9914bfac676de085440491528f00R381
So the info whether it `isIdempotent` or not needs to be extracted somehow.
Is it something that we should not do?
I don't see an alternative way of getting that info.
We can do:
```
MD5Digest id = MD5Digest.wrap(CBUtil.readBytes(body));
QueryHandler.Prepared prepared = QueryProcessor.instance.getPrepared(org.apache.cassandra.utils.MD5Digest.compute(id.bytes));
``` 
but it returns `QueryHandler.Prepared` NOT `ResultMessage.Prepared` (that we could cast to `PreparedWithIdempotent`).

Reply:
We decided that it is safe to return hard-coded `false` here because it is not used. 
done in: https://github.com/stargate/stargate/pull/1194/commits/8eddd4aadc9386ae9a244cb9cc2770bd76fde73b